Source/ExecutionEngine/ExecutionEngine.cs

```diff
@@ -918,26 +790,86 @@ public void Inline(Program program)
+        }
+
+        try {
+          ImplIdToCancellationTokenSource.AddOrUpdate(id, cts, (k, ov) => cts);
+
+          var coreTask = new Task<VerificationResult>(() => VerifyImplementation(program, stats, er, requestId,
+            extractLoopMappingInfo, implementation,
+            programId).Result, cts.Token, TaskCreationOptions.None);
+
+          coreTask.Start(LargeStackScheduler);
+          var verificationResult = await coreTask.WaitAsync(CancellationToken.None);
+          var output = verificationResult.Process(Options.Printer, this, stats, er, implementation);
+          outputCollector.Add(index, output);
+          outputCollector.WriteMoreOutput();
+          return verificationResult;
+        }
+        finally
+        {
+          ImplIdToCancellationTokenSource.TryRemove(id, out old);
+        }
       }
+    }
 
-      if (0 <= Options.VerifySnapshots && Options.TraceCachingForBenchmarking)
-      {
+    private Implementation[] GetPrioritizedImplementations(Program program)
+    {
+      // var impls = program.Implementations.Where(
+      //   impl => impl != null && Options.UserWantsToCheckRoutine(cce.NonNull(impl.Name)) &&
+      //           !impl.IsSkipVerification(Options));
+      //
+      // // operate on a stable copy, in case it gets updated while we're running
+      // Implementation[] stablePrioritizedImpls = null;
+      // if (0 < Options.VerifySnapshots)
+      // {
+      //   OtherDefinitionAxiomsCollector.Collect(Options, program.Axioms);
+      //   DependencyCollector.Collect(Options, program);
+      //   stablePrioritizedImpls = impls.OrderByDescending(
+      //     impl => impl.Priority != 1 ? impl.Priority : Cache.VerificationPriority(impl, Options.RunDiagnosticsOnTimeout)).ToArray();
+      // }
+      // else
+      // {
+      //   stablePrioritizedImpls = impls.OrderByDescending(impl => impl.Priority).ToArray();
+      // }
+      var impls = program.Implementations.Where(
+        impl => impl != null && Options.UserWantsToCheckRoutine(cce.NonNull(impl.Name)) &&
+                !impl.IsSkipVerification(Options));
+
+      // operate on a stable copy, in case it gets updated while we're running
+      Implementation[] stablePrioritizedImpls = null;
+      if (0 < Options.VerifySnapshots) {
+        OtherDefinitionAxiomsCollector.Collect(Options, program.Axioms);
+        DependencyCollector.Collect(Options, program);
+        stablePrioritizedImpls = impls.OrderByDescending(
+          impl => impl.Priority != 1 ? impl.Priority : Cache.VerificationPriority(impl, Options.RunDiagnosticsOnTimeout)).ToArray();
+      } else {
+        stablePrioritizedImpls = impls.OrderByDescending(impl => impl.Priority).ToArray();
+      }
+
+      return stablePrioritizedImpls;
+    }
+
+    private void TraceCachingForBenchmarking(PipelineStatistics stats,
+      string requestId, DateTime start)
+    {
+      if (0 <= Options.VerifySnapshots && Options.TraceCachingForBenchmarking) {
```

Comment:
`0 <=` is surprising (was expecting `0 <`), though I realize this isn't from your code.